exercises/book-store/src/example/java/BookStore.java

```diff
@@ -0,0 +1,62 @@
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
+public class BookStore{
+	public static double CalculateTotalCost(ArrayList<Integer> books ){
+		return CalculateTotalCost(books,0);
+	}
+
+	public static double CalculateTotalCost (ArrayList<Integer> books,double priceSoFar ){
+		double minPrice = Double.MAX_VALUE;
+
+		if(books.size() == 0)
+			return priceSoFar;
+
+		ArrayList<Integer> groups = (ArrayList<Integer>) books.stream().distinct().collect(Collectors.toList());
+
+		
+		double price = 0;
+
+		for(int i = 0;i<groups.size();i++){
+			books.remove(groups.get(i));	
+		}
+		
+		
+		try {
+			price = CalculateTotalCost(books,priceSoFar + CostPerGroup(groups.size()));
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+		minPrice = Math.min(minPrice, price);
+		return minPrice;
+
+
+	}
+
+	private static double CostPerGroup(int groupSize) throws Exception{
+		double discountPercentage;
+		switch (groupSize)
+		{
+		case 1:
+			discountPercentage = 0;
+			break;
+		case 2:
+			discountPercentage = 5;
+			break;
+		case 3:
+			discountPercentage = 10;
+			break;
+		case 4:
+			discountPercentage = 20;
+			break;
+		case 5:
+			discountPercentage = 25;
+			break;
```

Comment:
Sorry, I made an error!  I thought the discount tiers were linear by `groupSize`, and they are not.  My proposal and the file you uploaded both have a problem because of that.  However, a solution like the PHP one you mentioned would work nicely:

```Java
private static double[] DISCOUNT_TIERS = {0,5,10,20,25};   //declare at class level

private double CostPerGroup(int groupSize) throws Exception{
     if (groupSize < 1 || groupSize > 5){
        throw new Exception("Invalid group size : " + groupSize );
     }
    return bookPrice * groupSize * (100 - DISCOUNT_TIERS[groupSize-1])/100;
}
```